### ISSUE TITLE
linter: `no-unused-expressions` false positives

### ISSUE THREAD

**Issue by skovhus** [oxc-project/oxc#14275]
After `no-unused-expression` was moved to the correctness rule, it seems problematic that it reports that many false positives for code paths where you check a truth value and then execute a function.

```
  × eslint(no-unused-expressions): Expected expression to be used
    ╭─[myModal.ts]
 93 │       setLifecycle(Lifecycle.closed);
 94 │       props.isOpen && onRequestClose();
```

Note that the ESLint rule has the same issue.

**Comment by camc314** [comment]
This is the intended behaviour of the rule, as `props.isOpen && onRequestClose()` is a logical expression, and the value of the resulting logical expression is unused.

To change this behaviour, the rule can be configured with
```
{
    "rules": {
        "no-unused-expressions": ["error", { "allowShortCircuit": true }]
    }
}
```

This will allow code such as:
```
foo && bar()
```

**Comment by skovhus** [comment]
Ah, should have read up on the config. Thanks!
